### DISCUSSION TITLE
Ommission of an Argon2 password hashing function that encodes the result and takes a salt

### DISCUSSION THREAD

**Discussion by tchoutri** [jedisct1/libsodium#1230] [Q&A]
Hi! I am still writing my bindings to libsodium, and I realised that there is no documented way in the website to hash a password with the `crypto_pwhash` API in a way that encodes the result *and* gives control to the user over the salt. Not sure if this is a pattern that is enforced by design, and I wanted to have some clarification as to why `crypto_pwhash_str` didn't take a salt, whilst `crypto_pwhash` does. :)

Cheers!

**Comment by jedisct1** [comment]
Essentially because I can't come up with any valid use case for this.

Deterministic hashing is useful for key derivation. But when encrypting a password, what would be the point? The resulting string already encodes the salt, so there's no need to store it separately.
